index.js

```diff
@@ -228,14 +253,32 @@ async function validateCredentials(expectedAccessKeyId) {
   }
 }
 
-function getStsClient(region) {
+function getStsClient(region, credentials) {
   return new aws.STS({
     region,
     stsRegionalEndpoints: 'regional',
-    customUserAgent: USER_AGENT
+    customUserAgent: USER_AGENT,
+    credentials,
   });
 }
 
+function dropCredentials() {
+  core.debug('Dropping existing AWS credentials set in the environment')
+
+  const envs = [
+    'AWS_DEFAULT_REGION',
+    'AWS_REGION',
+    'AWS_ACCESS_KEY_ID',
+    'AWS_SECRET_ACCESS_KEY',
+    'AWS_SESSION_TOKEN',
+  ]
+
+  for (const e of envs) {
+    core.exportVariable(e, '');
+    delete process.env[e]
```

Comment:
The line `delete process.env[e]` seems redundant to me. `core.exportVariable` sets these values to empty, why would we need to delete them altogether in the second line? 

For reference, here's how exportVariable is implemented under the hood https://github.com/actions/toolkit/blob/b00a9fd033f4b30f2355acd212f531ecbbb9b38f/packages/core/src/core.ts#L83-L93